afterpay/src/main/kotlin/com/afterpay/android/internal/AfterpayInstalment.kt

```diff
@@ -0,0 +1,40 @@
+package com.afterpay.android.internal
+
+import com.afterpay.android.Afterpay
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.text.NumberFormat
+
+internal sealed class AfterpayInstalment {
+    data class Available(
+        val instalmentAmount: String
+    ) : AfterpayInstalment()
+
+    data class NotAvailable(
+        val minimumAmount: String?,
+        val maximumAmount: String
+    ) : AfterpayInstalment()
+
+    object NoConfiguration : AfterpayInstalment()
+
+    companion object {
+        fun of(totalCost: BigDecimal): AfterpayInstalment {
+            val configuration = Afterpay.configuration ?: return NoConfiguration
+
+            val currencyFormatter = NumberFormat.getCurrencyInstance().apply {
+                currency = configuration.currency
+            }
+
+            val minimumAmount = configuration.minimumAmount ?: BigDecimal.ZERO
+            if (totalCost < minimumAmount || totalCost > configuration.maximumAmount) {
+                return NotAvailable(
+                    minimumAmount = configuration.minimumAmount.let(currencyFormatter::format),
+                    maximumAmount = currencyFormatter.format(configuration.maximumAmount)
+                )
+            }
+
+            val instalment = (totalCost / 4.toBigDecimal()).setScale(2, RoundingMode.HALF_EVEN)
```

Comment:
I wish we were able to prove this is consistent with what afterpay does on their side. However it'll only ever be off by the smallest of amounts

Reply:
Yeah I figured this will be the closest we can get and should only be a cent off at the maximum.